examples/python/src/hello-world.py

```diff
@@ -0,0 +1,212 @@
+                    {
+                        'description': 'test command',
+                        'hint': '[This is testing command]',
+                        # this will be the command displayed to user as /second-command
+                        'label': 'second-command',
+                        'icon': 'icon.png',
+                        'bindings': [
+                            {
+                                'location': 'send',
+                                'label': 'send',
+                                'form': FORM
+                            },
+                        ],
+                    }
+                ],
+            },
+            {
+                'location': '/channel_header',
+                'bindings': [
+                    {
+                        'location': 'send-button',
+                        'icon': 'icon.png',
+                        'label': 'send hello message',
+                        'form': FORM,
+                    },
+                ],
+            },
+        ],
+    }
+
+
+@app.route('/ping', methods=['POST'])
+def on_ping() -> dict:
+    logging.debug('ping...')
+    return {'type': 'ok'}
+
+
+@app.route('/install', methods=['GET', 'POST'])
+def on_install() -> dict:
+    print(f'on_install called with payload , {request.args}, {request.data}', flush=True)
+    _subscribe_team_join(request.json['context'])
+    return {'type': 'ok', 'data': []}
+
+
+@app.route('/first_command', methods=['POST'])
+def on_first_command():
+    print(f'/first_command called ')
+    response_message = 'Hello! response from /first_command'
+    return {'type': 'ok', 'text': response_message}
+
+
+@app.route('/bot_joined_team', methods=['GET', 'POST'])
+def on_bot_joined_team() -> dict:
+    context = request.json['context']
+    logging.info(
+        f'bot_joined_team event received for site:{context["mattermost_site_url"]}, '
+        f'team:{context["team"]["id"]} name:{context["team"]["name"]} '
+        f'{request.args} {request.data}'
+    )
+    # Here one can subscribe to channel_joined/left events as these required team_id now to be subscribed,
+    # hence use the team_id received in the event and make a call for subscribing to channel_joined/left events.
+    # Also supply {'team_id': team_id} in the request body of the subscription
+    # {
+    #    'subject': 'bot_joined_team',
+    #    'call': {
+    #        'path': '/bot_joined_team',
+    #         'expand': {
+    #             'app': 'all',
+    #             'team': 'all'
+    #         }
+    #    },
+    #    'team_id': 'team_id'   # get this team_id when bot_joined_team event occurs
+    # }
+    return {'type': 'ok', 'data': []}
+
+
+# Subscribing to events. For example, Subscribe to 'bot_joined_team' event
+def _subscribe_team_join(context: dict) -> None:
+    site_url = context['mattermost_site_url']
+    bot_access_token = context['bot_access_token']
+    url = site_url + 'plugins/com.mattermost.apps/api/v1/subscribe'
```

Comment:
Should we do `path.join` here instead? Does the site URL always end in a slash?

Reply:
Use of `path.join` could result in different result based on the platform running this code. 
For example, in Windows that would result into `\` while running on Mac system, that would become `/`. 

We need to use some url specific libs for this.

Reply:
done.